src/Symfony/Component/HttpClient/Response/StreamWrapper.php

```diff
@@ -178,6 +178,56 @@ public function stream_read(int $count)
         return '';
     }
 
+    public function stream_set_option(int $option, int $arg1, ?int $arg2): bool
+    {
+        if (null === $this->handle || 'stream' !== get_resource_type($this->handle)) {
+            trigger_error(sprintf('The "$handle" property of "%s" need to be a stream.', __CLASS__), E_USER_WARNING);
+
+            return false;
+        }
+
+        switch ($option) {
+            case STREAM_OPTION_BLOCKING:
+                return $this->stream_set_blocking($this->handle, $arg1);
+                break;
+            case STREAM_OPTION_READ_TIMEOUT:
+                return $this->stream_set_timeout($this->handle, $arg1, $arg2);
+                break;
+            case STREAM_OPTION_WRITE_BUFFER:
+                return $this->stream_set_write_buffer($this->handle, $arg1);
+                break;
+            default:
+                trigger_error(sprintf('The option "%s" is unknown for "stream_set_option" method', $option), E_USER_WARNING);
+
+                return false;
+        }
+    }
+
+    public function stream_set_blocking($stream, bool $mode): bool
+    {
+        if (true === $mode) {
+            trigger_error(sprintf('Blocking mode is not supported yet with "%s" class.', __CLASS__), E_USER_WARNING);
+
+            return false;
+        }
+
+        return true;
+    }
+
+    public function stream_set_write_buffer($stream, int $buffer): int
+    {
+        trigger_error(sprintf('Writting with "%s" class, is not supported yet.', __CLASS__), E_USER_WARNING);
+
+        return false;
+    }
+
+    public function stream_set_timeout($stream, int $seconds, int $microseconds = 0): bool
```

Comment:
Those 3 functions are useless and shoud be removed. `stream_set_blocking` `stream_set_write_buffer`  and `stream_set_timeout` 

Or at least they should not be public.

Reply:
They are now private. IMHO it's better to have specific little functions to handle each case.

The logic of `stream_set_option` function is already disturbing. ^^